src/towncrier/_project.py

```diff
@@ -5,17 +5,24 @@
 Responsible for getting the version and name from a project.
 """
 
-
 from __future__ import annotations
 
 import sys
 
 from importlib import import_module
+from importlib.metadata import version as metadata_version
 from types import ModuleType
+from typing import Any
 
 from incremental import Version as IncrementalVersion
 
 
+try:
+    from importlib.metadata import packages_distributions
+except ImportError:
+    from importlib_metadata import packages_distributions  # type: ignore
```

Comment:
Maybe add a comment.
This should help with a future cleanup, once 3.9 is no longer supported

```suggestion
    # Fallback on non-stdlib version for
    # Python 3.9 or older.
    from importlib_metadata import packages_distributions  # type: ignore
```